select_assignment2_flicks/Model/Movie.swift

```diff
@@ -41,6 +42,39 @@ struct Movie {
     
     var voteAverage: Double = 0
     
+    var myRootRef = Firebase(url: Constants.FIREBASE_URL)
+    
+    func isFavoriteFireBase(onComplete: (list: [String], value: Bool) -> Void) {
+        let deviceRef = myRootRef.childByAppendingPath(UIDevice.currentDevice().identifierForVendor!.UUIDString)
+        
+        let favoritesRef = deviceRef.childByAppendingPath("favorites")
+        
+        favoritesRef.observeEventType(.Value, withBlock: { snapshot in
+            if let favorites = snapshot.value as? [String] {
+                onComplete(list: favorites, value: favorites.indexOf(self.id) >= 0)
+                return
+            }
+            onComplete(list: [], value: false)
+        })
+    }
+    
+    func setFavoriteFireBase(onComplete: () -> Void) {
+        isFavoriteFireBase { (list, value) in
+            guard !value else {
+                return
+            }
+            var newList = list
+            newList.append(self.id)
+            let deviceRef = self.myRootRef.childByAppendingPath(UIDevice.currentDevice().identifierForVendor!.UUIDString)
+            
+            let favoritesRef = deviceRef.childByAppendingPath("favorites")
+            
+            favoritesRef.setValue(newList) { (error, firebase) in
```

Comment:
very small thing here, if you don't use the parameters, you should name the parameters to the closure as `_`